src/Sylius/Bundle/ResourceBundle/Form/EventSubscriber/AddCodeFormSubscriber.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Form\EventSubscriber;
+
+use Sylius\Component\Resource\Model\CodeAwareInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormEvent;
+use Symfony\Component\Form\FormEvents;
+use Symfony\Component\Intl\Exception\UnexpectedTypeException;
+
+/**
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+class AddCodeFormSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @return array
+     */
+    public static function getSubscribedEvents()
+    {
+        return array(
+            FormEvents::PRE_SET_DATA => 'preSetData',
+        );
+    }
+
+    /**
+     * @param FormEvent $event
+     */
+    public function preSetData(FormEvent $event)
+    {
+        $resource = $event->getData();
+
+        if (!$resource instanceof CodeAwareInterface) {
```

Comment:
@pamil @pjedrzejewski I thought that when the `create` form will be filled and the validation fails because e.g. I left blank name the `code` will not be able to be modified. That's not the case since in the ResourceController we first set a new object on the form (then the subscriber is invoked) and after that the data from the request is binded, so it will work fine, but if we will change this workflow we can introduce the unwanted behavior I described.

Highlighting this because I don't know what is the workflow in the new ResourceController ;)